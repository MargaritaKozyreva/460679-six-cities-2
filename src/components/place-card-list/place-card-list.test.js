import React from "react";
import PlaceCardList from "./place-card-list.jsx";
import renderer from "react-test-renderer";

describe(`initial PlaceCardList`, () => {

  const onTitleClickHandler = jest.fn();

  const mockProps = {
    cards: [
      {
        type: `Private room`,
        offers: [
          {
            id: 4,
            title: `Nice, cozy, warm big bed apartment`,
            link: `#`,
            price: 170,
            img: `img/apartment-01.jpg`,
            raiting: 3
          }
        ]
      },
      {
        type: `Private room`,
        offers: [
          {
            id: 4,
            title: `Nice, cozy, warm big bed apartment`,
            link: `#`,
            price: 170,
            img: `img/apartment-01.jpg`,
            raiting: 3
          }
        ]
      }
    ],
    onTitleClickHandler
  };

  it(`PlaceCardList renders correctly`, () => {
    const tree = renderer.create(<PlaceCardList {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
