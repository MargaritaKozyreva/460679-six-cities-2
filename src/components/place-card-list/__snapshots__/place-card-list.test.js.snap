// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial PlaceCardList PlaceCardList renders correctly 1`] = `
Array [
  <article
    className="cities__place-card place-card"
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a>
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            className="visually-hidden"
          />
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[MockFunction]}
      >
        <a
          href="/details"
        />
      </h2>
      <p
        className="place-card__type"
      >
        Private room
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    onMouseOver={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a>
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            className="visually-hidden"
          />
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[MockFunction]}
      >
        <a
          href="/details"
        />
      </h2>
      <p
        className="place-card__type"
      >
        Private room
      </p>
    </div>
  </article>,
]
`;
